contracts/ExampleTrustcoin2.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity ^0.4.8;
+
+import './lib/ERC20.sol';
+import './lib/SafeMath.sol';
+import './Trustcoin.sol'
+
+contract Trustcoin2 is ERC20, SafeMath {
+
+  string public name = 'Trustcoin2';
+  uint8 public decimals = 18;
+  string public symbol = 'TRST2';
+  string public version = 'TRST2.0';
+  uint256 public totalSupply;
+  address public oldToken;
+  bool public allowOldMigrations;
+  uint256 public allowOldMigrationsUntil;
+
+  mapping(address => uint) balances;
+  mapping (address => mapping (address => uint)) allowed;
+
+  bool migrationActive;
+  address migrationMaster;
+
+  event Transfer(address from, address to, uint256 value);
+  event Approval(address from, address to, uint256 value);
+  event Migrate(address owner, uint256 value);
+  event MigrateSuccess(address owner, uint256 value);
+  event MigrationFinalized();
+
+  function Trustcoin2(address _oldToken, address _migrationMaster, bool _allowOldMigrations, uint256 _allowOldMigrationsUntil) {
```

Comment:
Specifically
- I find it hard to understand when `allowOldMigrations` might be false.
- Having checks on `allowOldMigrationsUntil` seems like unnecessary code. When we deploy the contract with an "until date", users will be able to examine the whole contract and see if the date is far enough in the future. I can't see how code checks that check a constructor constant for one-off contract could help.